zip-reduce-shielded_tx_fee.rst

```diff
@@ -0,0 +1,157 @@
+new information to users who might use Zcash for e-commerce
+and app interactions that might result in several transactions each day.
+
+Privacy must not cost a premium. The cost of 10 shielded transactions
+buys 1GB of mobile data `in India today <https://www.cable.co.uk/mobiles/worldwide-data-pricing/>`.
+
+Zcash users must be able to do more with their ZEC balance
+than worry about paying the premium for shielded transactions.
+
+
+Requirements for consensus
+-------------------------
+
+The change to the conventional transaction fees must be undertaken sooner
+as it gets difficult to gain consensus with the growth in the network
+of wallets, exchanges, miners and third parties involved.
+
+The following parties need to be part of the consensus:
+
+* A guarantee from mining groups is required to include the lowered default fee transactions in the next block.
+* Wallet developers need to provide commitment to update the software to use the new fee.
+* Zcash documentation and community outreach must be undertaken to make the change known.
+
+
+Security and privacy considerations
+-----------
+
+Unique transaction fees may reveal specific users or wallets or wallet versions which reduces privacy for those specific users and the rest of the network.
+hence this change must be accepted by majority, if not all popular
+shielded transaction software providers before announcing the change.
+
+Varying/unique fees are bad for privacy, for the short term before blocks get full,
+it’s fine for everyone to use a constant fee, as long as that is enough to compensate miners for including the transaction. [#nathan-1]_
+
+Long term, the issue of fees needs to be re-visited in separate future proposals as the blocks start getting consistently full.
+New ZIPs with flexible fees, such as [#ian-1]_, along with scaling solutions need to be evaluated and applied.
+
+Denial Of Service Vulnerability
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+A transaction-rate-based denial of service attack occurs when an attacker generates enough transactions over a window of time to prevent legitimate transactions from being mined, or to hinder syncing blocks for full nodes or miners.
+
+There are two primary protections to this kind of attack in Zcash: the block size limit, and variable transaction fees. The block size limit ensures that full nodes and miners can sync blocks even if they are completely full. However it does not protect users sending legitimate transactions to have their transactions confirmed in a timely manner.
+
+Variable fees can mitigate this kind of denial of service because in there are more transactions available than can fit into a single block, a miner is assumed to choose the transactions that pay the highest fees. If legitimate wallets increase their fees during this condition, the attacker must also increase the fees of their transactions. This imposes a growing and ongoing cost to the attacker which limits the time window they can continue the attack.
```

Comment:
```suggestion
Variable fees could mitigate this kind of denial of service: if there are more
transactions available than can fit into a single block, then a miner will typically
choose the transactions that pay the highest fees. If legitimate wallets were to
increase their fees during this condition, the attacker would also increase the
fees of their transactions. It is sometimes argued that this would impose a cost
to the attacker that would limit the time window they can continue the attack.
However, there is little evidence that the actual costs involved would be a sufficient
disincentive.
```